server/tests/e2e/test_stat_keeper.py

```diff
@@ -0,0 +1,1730 @@
+    expect_players_enabled(page, home_line)
+    expect_players_enabled(page, away_line)
+    expect_players_disabled(page, home_bench)
+    expect_players_disabled(page, away_bench)
+
+    # all actions disabled until a player is picked to start
+    expect_button_disabled(page, "Pull")
+    expect_button_disabled(page, "Point!")
+    expect_button_disabled(page, "Drop")
+    expect_button_disabled(page, "Throwaway")
+    expect_button_disabled(page, "D (Block)")
+    expect_button_disabled(page, "Catch D")
+    # nothing has happened yet. nothing to undo
+    expect_button_disabled(page, "Undo")
+
+    # player is chosen
+    # away is pulling
+    click_button(page, "Owen Lumley")
+    expect_players_disabled(page, home_line)
+    expect_players_disabled(page, away_line)
+    # only pull and undo enabled
+    expect_button_enabled(page, "Pull")
+    expect_button_disabled(page, "Point!")
+    expect_button_disabled(page, "Drop")
+    expect_button_disabled(page, "Throwaway")
+    expect_button_disabled(page, "D (Block)")
+    expect_button_disabled(page, "Catch D")
+    expect_button_enabled(page, "Undo")
+
+    # pull has to be next (or undo)
+    click_button(page, "Pull")
+    expect_players_enabled(page, home_line)
+    expect_players_disabled(page, away_line)
+    expect_button_disabled(page, "pull")
+    expect_button_disabled(page, "point!")
+    expect_button_disabled(page, "drop")
+    expect_button_disabled(page, "throwaway")
+    expect_button_disabled(page, "d (block)")
+    expect_button_disabled(page, "catch d")
+
+    # player receives the pull or picks it up
+    # they can drop it or throw it away
+    click_button(page, "Brian Kells")
+    expect_players_enabled(page, [p for p in home_line if p != "Brian Kells"])
+    expect_players_disabled(page, away_line)
+    expect_button_disabled(page, "pull")
+    expect_button_disabled(page, "point!")
+    expect_button_enabled(page, "drop")
+    expect_button_enabled(page, "throwaway")
+    expect_button_disabled(page, "d (block)")
+    expect_button_disabled(page, "catch d")
+
+    # first pass
+    click_button(page, "Ashlin Kelly")
+    expect_players_enabled(page, [p for p in home_line if p != "Ashlin Kelly"])
+    expect_players_disabled(page, away_line)
+    expect_button_disabled(page, "pull")
+    expect_button_disabled(page, "point!")  # point disabled (QuebecVariant)
+    expect_button_enabled(page, "drop")
+    expect_button_enabled(page, "throwaway")
+    expect_button_disabled(page, "d (block)")
+    expect_button_disabled(page, "catch d")
+
+    # turnover
+    click_button(page, "Throwaway")
+    expect_players_disabled(page, home_line)
+    expect_players_enabled(page, away_line)
+    expect_button_disabled(page, "pull")
+    expect_button_disabled(page, "point!")
+    expect_button_disabled(page, "drop")
+    expect_button_disabled(page, "throwaway")
+    expect_button_disabled(page, "d (block)")
+    expect_button_disabled(page, "catch d")
```

Comment:
some inconsistent capitalization here. This could probably be improved with a method that takes only the buttons we expect to be enabled and explicitly confirms the others are disabled